File: 백준/Silver/1965. 상자넣기/상자넣기.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
n = int(input())
numbers = list(map(int, input().split()))
res = [1 for _ in range(n)]
for i in range(1, n):
    for j in range(i):
        if numbers[i]>numbers[j]:
            res[i] = max(res[i], res[j]+1)
print(max(res))

File: 백준/Silver/1965. 상자넣기/상자넣기.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	var n, max int
	fmt.Fscanf(reader, "%d\n", &n)
	var numbers = make([]int, n)
	var res = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &numbers[i])
		res[i] = 1
	}
	for i := 1; i < n; i++ {
		for j := 0; j < i; j++ {
			if numbers[i] > numbers[j] {
				if res[i] < res[j]+1 {
					res[i] = res[j] + 1
				}
			}
		}
	}
	for _, v := range res {
		if v > max {
			max = v
		}
	}
	fmt.Fprintf(writer, "%d\n", max)
}
